client/client.go

```diff
@@ -309,7 +309,7 @@ var (
 )
 
 // NewClient is used to create a new client from the given configuration
-func NewClient(cfg *config.Config, consulCatalog consul.CatalogAPI, consulProxies consulApi.SupportedProxiesAPI, consulService consulApi.ConsulServiceAPI) (*Client, error) {
+func NewClient(cfg *config.Config, consulCatalog consul.CatalogAPI, consulProxies consulApi.SupportedProxiesAPI, consulService consulApi.ConsulServiceAPI, rpcs map[string]interface{}) (*Client, error) {
```

Comment:
I'd suggest documenting the `rpcs` parameters, as `interface{}` is very ambiguous.  I'd link somewhere to https://golang.org/pkg/net/rpc/#Server.Register .